src/ui/remotesignalsources.cpp

```diff
@@ -0,0 +1,238 @@
+    : filters(filters) {
+    filters.onChange.emplace_back(myOnChange);
+    update();
+}
+SignalList::~SignalList() {
+    auto it = std::find_if(filters.onChange.begin(), filters.onChange.end(), [this](auto e) { return e == myOnChange; });
+    assert(it != filters.onChange.end());
+    filters.onChange.erase(it);
+
+    // TODO test this
+}
+void SignalList::update() {
+    opencmw::service::dns::QueryEntry queryEntry;
+
+    refl::util::for_each(refl::reflect(queryEntry).members, [&](auto member) {
+        // TODO maybe pick the last instead of the first
+
+        auto it = std::find_if(filters.begin(), filters.end(), [&member](const auto &f) {
+            return f.selectedField() == refl::descriptor::get_display_name(member);
+        });
+        // we pick the first
+        auto &strValue = it->filterText;
+        if (it != filters.end() && strValue != "") {
+            if constexpr (std::is_integral_v<std::remove_cvref_t<decltype(member(queryEntry))>>)
+                member(queryEntry) = std::atoi(strValue.c_str());
+            else if constexpr (std::is_floating_point_v<std::remove_cvref_t<decltype(member(queryEntry))>>)
+                member(queryEntry) = std::atof(strValue.c_str());
+            else
+                member(queryEntry) = strValue;
+        }
+    });
+    try {
+        std::unique_lock l{ signalsMutex };
+        dnsClient.querySignalsAsync([this](const std::vector<opencmw::service::dns::Entry> &entries) {
+            signals = entries;
+        },
+                queryEntry);
+    } catch (const std::exception &e) {
+        std::cerr << "Error loading signals: " << e.what() << std::endl;
+    }
+}
+void SignalList::drawElements() {
+    ImGui::BeginTable("Signals", refl::reflect<opencmw::service::dns::QueryEntry>().members.size + 1, ImGuiTableFlags_BordersInnerV);
+
+    ImGui::TableHeader("SignalsHeader");
+    refl::util::for_each(refl::reflect<opencmw::service::dns::QueryEntry>().members, [](auto m) {
+        ImGui::TableSetupColumn(static_cast<const char *>(m.name));
+    });
+    ImGui::TableSetupColumn("Add Signal");
+    ImGui::TableHeadersRow();
+    {
+        std::unique_lock l{ signalsMutex };
+        std::for_each(signals.begin(), signals.end(), [this, idx = 0](const auto &e) mutable { drawElement(e, idx++); });
+    }
+    ImGui::EndTable();
+}
+void SignalList::drawElement(const opencmw::service::dns::Entry &entry, int idx) {
+    ImGui::TableNextRow();
+    refl::util::for_each(refl::reflect<opencmw::service::dns::QueryEntry>().members, [&entry](auto m) {
+        auto value = m(entry);
+        ImGui::TableNextColumn();
+        if constexpr (std::is_integral_v<decltype(value)> || std::is_floating_point_v<decltype(value)>) {
+            ImGui::TextUnformatted(std::to_string(m(entry)).c_str());
+        } else {
+            ImGui::TextUnformatted(value.c_str());
+        }
+    });
+    ImGui::TableNextColumn();
+    if (ImGui::Button(("+##" + std::to_string(idx)).c_str())) {
```

Comment:
Just a remark since I came upon this also in another project, instead of constructing identity strings, you can use `ImGui::PushID(idx)`/`ImGui::PopID()` around each table row.